src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Logging/SuppressionEngine.cs

```diff
@@ -30,7 +30,10 @@ public class SuppressionEngine : ISuppressionEngine
             string? noWarn = null,
             bool baselineAllErrors = false)
         {
-            _validationSuppressions = ParseSuppressionFile(suppressionFile);
+            // Don't read from the passed in suppression file if errors will be baselined and the file doesn't exist.
+            _validationSuppressions = baselineAllErrors && !File.Exists(suppressionFile) ?
+                new HashSet<Suppression>() :
+                ParseSuppressionFile(suppressionFile);
```

Comment:
Maybe the `File.Exists` check could be added to the `ParseSuppressionFile` method at the beginning, and if there is no file, let it return an empty `HashSet`.

I suggest this because if there are other locations where the `ParseSuppressionFile` method needs to be used in the future, you would have to add similar `File.Exists` checks there.